### PR TITLE
Cleanup whole codebase

### PR DESCRIPTION
**Pull request by Pringlers**
Please squash merge it to remove inconsistent commit messages later

# Goals 
This PR is intended to clean up the entire codebase.
- Create, modify, or remove the classes for better maintainability
- Adopt more efficient logic
- Remove duplicated code
- Write idiomatic TypeScript

# Non-Goals
- Change website appearance
- Create API v3
- Change DB structure

### CODE REVIEW

File: utils/Query.ts [hunk 1]
```diff
@@ -45,45 +45,63 @@ async function getBot(id: string, topLevel=true):Promise<Bot> {
...
-			res[0].owners = await Promise.all(
-				res[0].owners.map(async (u: string) => await get._rawUser.load(u))
-			)
-			res[0].owners = res[0].owners.filter((el: User | null) => el).map((row: User) => ({ ...row }))
-		}
 
-		await knex('bots').update({ name: discordBot.username }).where({ id })
+	if (!botRow[0]) return null
+
+	// FIXME: Handle this properly
```
Comment (skinmaker1345) [inline utils/Query.ts:51]: you may assert; further checks are being considered to be implemented in near future.

File: utils/Query.ts [hunk 2]
```diff
@@ -103,69 +121,102 @@ async function getServer(id: string, topLevel=true): Promise<Server> {
...
+		.select(['id'])
+		.where('owners', 'like', `%${id}%`)
+		.orderBy('date', 'asc')
+	const ownedServers = await knex('servers')
+		.select(['id'])
+		.where('owners', 'like', `%${id}%`)
+		.orderBy('date', 'asc')
+	const discordUser = await get.discord.user.load(id)
+
+	// FIXME: Missing type
```
Comment (skinmaker1345) [inline utils/Query.ts:200]: each props are typed as like Bot[] | string[], please leave it as it is

File: utils/Query.ts [hunk 3]
```diff
@@ -45,45 +45,63 @@ async function getBot(id: string, topLevel=true):Promise<Bot> {
...
+	if (botMember.user.flags.has(UserFlags.BotHTTPInteractions)) {
+		status = 'online'
+	} else if (!botMember.presence) {
+		status = 'offline'
+	} else {
+		const isStreaming = botMember.presence.activities.some(activity => activity.type === ActivityType.Streaming)
+		status = isStreaming ? 'streaming' : botMember.presence.status
+	}
 
+	// FIXME: Missing type
```
Comment (skinmaker1345) [inline utils/Query.ts:79]: ditto

File: utils/Query.ts [hunk 4]
```diff
@@ -103,69 +121,102 @@ async function getServer(id: string, topLevel=true): Promise<Server> {
...
 	}
-	return res[0] ?? null
+
+	// Delete deprecated fields
+	delete serverRow[0].owners
+
+	// FIXME: Type mismatch
+	let owner: any = data?.owner ?? null
+	let bots: any = data?.bots ?? null
+	if(topLevel) {
```
Comment (skinmaker1345) [inline utils/Query.ts:152]: this is the problem with typing, maybe you are okay with editing the type as User | string | null.

**Comment by Pringlers** [comment]
@skinmaker1345 There are more changes to be made in this PR, or do you want me to split this into multiple PRs?

**Comment by Pringlers** [comment]
Type rework fixed every relevant type error.

image

Most of the changes in components are as simple as removing type assertions (e.g. `(bot.users as User[])`).
Components that only operate on partial data now require the type of `XXX<true|false>`

**Comment by Pringlers** [comment]
`XXXTable` – 1:1 mapping of SQL query result. JSON fields are not parsed yet
`BaseXXX` – Same, but JSON fields are parsed
`XXX<Nested>` – `Nested` indicates whether it has fully resolved children or snowflakes.

**Comment by skinmaker1345** [comment]
Looks great, except
@Pringlers your code seems to fetch nested objects, always, even for raw objects

**Comment by skinmaker1345** [comment]
> image
> 
we are aware of that, what's the ts version you're running., by chance?

**Comment by Pringlers** [comment]
> your code seems to fetch nested objects, always, even for raw objects
Can you point out where it does? I moved code to resolve children into `getXXX`, and `_rawXXX` DataLoaders now call `getNestedXXX(id)` directly if you mean that

**Comment by skinmaker1345** [comment]
> Can you point out where it does? I moved code to resolve children into getXXX, and _rawXXX DataLoaders now call getNestedXXX(id) directly if you mean that
think i confused each other, you did refactor as expected, apologies.
> 4.9.5
using 4.6.x resolves the issue(if you're having trouble with it), 4.6.4 is the version which we specify to use

**Comment by skinmaker1345** [comment]
@Pringlers Do you have any works to be done?

**Comment by Pringlers** [comment]
Yes I do, but it might take long to get them done. So feel free to merge it if you want to work on top of these changes

**Comment by skinmaker1345** [comment]
We need to know what will be changed.We already have plans to rewrite, and we would like to limit your job since your works might be disposed.2023. 3. 7. 오후 5:31, Pringlers ***@***.***> 작성:﻿
Yes I do, but it might take long to get them done. So feel free to merge it if you want to work on top of these changes

**Comment by skinmaker1345** [comment]
Hi, I am afraid to tell that this pr should be closed as we are preparing to apply a formatter for entire codebase.

**Pull request closed without merging**